### ISSUE TITLE
Query Frontend: Incorrect cache keys for dynamic split interval

### ISSUE THREAD

**Issue by lachruzam** [thanos-io/thanos#7831]
**Thanos, Prometheus and Golang version used**:
quay.io/thanos/thanos:v0.36.1

**What happened**:
The interval used to generate cache keys differs from the interval used to split the original query.

for example, with the settings:
```
--query-range.min-split-interval=2h
--query-range.max-split-interval=96h
--query-range.horizontal-shards=12
```
a query with an 8h range and a 5m step is split using a 40m interval, while cache keys are generated using a 2h interval.

**What you expected to happen**:
Cache keys should be generated using the same interval as used for query splitting.

**How to reproduce it (as minimally and precisely as possible)**:
Run query-frontend using settings listed above.

**Full logs to relevant components**:

**Anything else we need to know**:

**Comment by lachruzam** [comment]
PR: https://github.com/thanos-io/thanos/pull/7832

**Comment by yeya24** [comment]
Fixed by #7832
